go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -807,3 +813,34 @@ func (bnc *BaseNetworkController) nodeZoneClusterChanged(oldNode, newNode *kapi.
 
 	return false
 }
+
+func (bnc *BaseNetworkController) findMigratablePodIPsForSubnets(subnets []*net.IPNet) ([][]*net.IPNet, error) {
+	ipSet := sets.New[string]()
+	ipsByPod := [][]*net.IPNet{}
+	liveMigratablePods, err := kubevirt.FindLiveMigratablePods(bnc.watchFactory)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, liveMigratablePod := range liveMigratablePods {
```

Comment:
Have you considered what to do with completed pods? I think the least evil is to allocate them but we do have a race condition here because we don't know if the pod handler has already processed the completed pods or not so the IPs we allocate for them here might never be de-allocated.

This could potentially be fixed by taping into the pod handler obj_retry to check if the pod completion has been processed with the necessary locking (locking which might turn out to be giant).

We might choose not to solve it given the corner case nature of the situation but I would at least add a big FIXME.

Pulling in @trozet for a second opinion.

Reply:
I think we should not allocate any pod ips that are completed, but maybe there is some scenario I'm missing...

Reply:
The opposite side of the race would be if we don't allocate a completed pod that has not yet been released, the the same IP might be allocated to a different pod and then when the completed pod releases we release an IP that is also used by a different pod.

But yeah, now that I think about it, not allocating them might be the least evil for now.

Reply:
still missing to filter out completed pods?

Reply:
Right, checking it.